main_gin.py

```diff
@@ -99,8 +100,21 @@ def set_seeds(seed):
 with open("train_valid_test_split_idxs_dict.pkl", "wb") as f:
     pickle.dump(split_idx, f)
 print("dumped train/valid/test split indices dict to train_valid_test_split_idxs_dict.pkl (note these will be randomized each run by default so you should save this file for reproducibility)")
-train_loader = DataLoader(dataset[split_idx["train"]], batch_size=config["batch_size"], shuffle=True)
-valid_loader = DataLoader(dataset[split_idx["valid"]], batch_size=config["batch_size"], shuffle=False)
+
+# note they are preshuffled and presplit into training/validation/test splits
+if args.hold_out_addl_data_from_train_set_as_dev_for_addl_generalization_check and (len(split_idx["train"]) < 2 * len(split_idx["valid"])):
+    raise Exception("Cannot use --hold_out_addl_data_from_train_set_as_dev_for_addl_generalization_check when validation set is greater than half the train set size due to insufficient data remaining.")
+
+train_split = split_idx["train"] if not args.hold_out_addl_data_from_train_set_as_dev_for_addl_generalization_check else split_idx["train"][len(split_idx["valid"]):]
+valid_split = split_idx["valid"]
+
+dev_split = None if not args.hold_out_addl_data_from_train_set_as_dev_for_addl_generalization_check else split_idx["train"][:len(split_idx["valid"])]# NOT TEST, this is taking some of the existing training split
```

Comment:
Note, the `--hold_out_addl_data_from_train_set_as_dev_for_addl_generalization_check` argument as added here would NOT ensure that the new split out of training set would contain any minimum number of the active molecules (they make up less than 0.2% of the examples in this data). For WNV PCBA 577 there are VERY FEW active molecules to split between the different sets and if this is not checked I was observing e.g. 7 molecules only in this hold out leading to really unreliable (high variance) assessments of generalization in this training data subset hold out. So making a change to ensure that the holdout from train of the same size as the validation set has the same number of active molecules as the validation set for fair comparison.

Reply:
See https://github.com/willy-b/tiny-GIN-for-PCBA-577-WNV/pull/10/files#r1797735363 for upcoming change.